test/CSStrikes.t.sol

```diff
@@ -0,0 +1,279 @@
+        new CSStrikes(address(0), oracle);
+    }
+
+    function test_initialize_RevertWhen_ZeroOracleAddress() public {
+        vm.expectRevert(ICSStrikes.ZeroOracleAddress.selector);
+        new CSStrikes(module, address(0));
+    }
+}
+
+contract CSStrikesTest is CSStrikesTestBase {
+    function setUp() public {
+        stranger = nextAddress("STRANGER");
+        oracle = nextAddress("ORACLE");
+        module = nextAddress("MODULE");
+
+        strikes = new CSStrikes(module, oracle);
+
+        tree = new MerkleTree();
+
+        vm.label(address(strikes), "STRIKES");
+    }
+
+    function test_hashLeaf() public assertInvariants {
+        uint256 noId = 42;
+        (bytes memory pubkey, ) = keysSignatures(1);
+        uint256[] memory strikesData = new uint256[](6);
+        strikesData[0] = 100500;
+
+        assertEq(
+            strikes.hashLeaf(noId, pubkey, strikesData),
+            0x01eba5ed7fb9c5ebb4262844c7125628afcbea15e57bc7f4dd0c80d34d633584
+        );
+    }
+
+    function test_verifyProof() public {
+        uint256 noId = 42;
+        (bytes memory pubkey, ) = keysSignatures(1);
+        uint256[] memory strikesData = new uint256[](6);
+        strikesData[0] = 100500;
+
+        tree.pushLeaf(abi.encode(noId, pubkey, strikesData));
+        tree.pushLeaf(abi.encode(noId + 1, pubkey, strikesData));
+
+        stdstore.target(address(strikes)).sig("treeRoot()").checked_write(
+            tree.root()
+        );
+
+        bytes32[] memory proof = tree.getProof(0);
+
+        strikes.verifyProof({
+            nodeOperatorId: noId,
+            pubkey: pubkey,
+            strikesData: strikesData,
+            proof: proof
+        });
+    }
+
+    function test_verifyProof_RevertWhen_InconsistentData() public {
+        uint256 noId = 42;
+        (bytes memory pubkey, ) = keysSignatures(1);
+        uint256[] memory strikesData = new uint256[](6);
+        strikesData[0] = 100500;
+
+        tree.pushLeaf(abi.encode(noId, pubkey, strikesData));
+        tree.pushLeaf(abi.encode(noId + 1, pubkey, strikesData));
+
+        stdstore.target(address(strikes)).sig("treeRoot()").checked_write(
+            tree.root()
+        );
+
+        bytes32[] memory proof = tree.getProof(1);
+
+        vm.expectRevert(ICSStrikes.InvalidProof.selector);
+        strikes.verifyProof({
+            nodeOperatorId: noId,
+            pubkey: pubkey,
+            strikesData: strikesData,
+            proof: proof
+        });
+    }
+
+    function test_verifyProof_RevertWhen_EmptyProof() public {
+        uint256 noId = 42;
+        (bytes memory pubkey, ) = keysSignatures(1);
+        uint256[] memory strikesData = new uint256[](6);
+        strikesData[0] = 100500;
+
+        // Put a vulnerable `treeRoot` to make sure we provided the valid, but empty proof.
+        stdstore.target(address(strikes)).sig("treeRoot()").checked_write(
+            strikes.hashLeaf(noId, pubkey, strikesData)
+        );
```

Comment:
I'm not sure if I get it
why it's vulnerable?

if you need to put any bytes here, it's better to put random bytes but not something that looks like a real value (tree leaf)

Reply:
just copy-paste